www/class/centreonService.class.php

```diff
@@ -224,48 +247,55 @@ public function getServiceName($sid)
     }
 
     /**
-     * Get Service alias
+     * Gets the service description of a service
      *
-     * @param int $sid
-     * @return string
+     * @param int[] $serviceIds
+     * @return array retArr
      */
-    public function getServicesDescr($sid = array())
+    public function getServicesDescr($serviceIds = []): array
     {
-        $arrayReturn = array();
-
-        if (!empty($sid)) {
-            $where = "";
-            foreach ($sid as $s) {
-                $tmp = explode("_", $s);
-                if (isset($tmp[0]) && isset($tmp[1])) {
-                    if ($where !== "") {
-                        $where .= " OR ";
-                    } else {
-                        $where .= " AND ( ";
+        $retArr = [];
+
+        if (!empty($serviceIds)) {
+            $where = '';
+            /* checking here that the array provided as parameter
+             * is exclusively made of integers (service ids)
+             */
+            $filteredSvcIds = $this->filteredArrayId($serviceIds);
+
+            if (count($filteredSvcIds) > 0) {
+                foreach ($filteredSvcIds as $s) {
+                    $tmp = explode("_", $s);
```

Comment:
why not get this results from the filteredArrayId method. As you've already used the explode. And need to check here too that the "_" character is available in each $s

Reply:
The only purpose of the `filteredArrayId` function is to filter the array provided as parameter of the undesired values (that does not match `([0-9]+)_([0-9]+)`). Once filtered we've made sure that values are pure ints.

Reply:
with last modifications, you should not need anymore the !empty check on line 272